### ISSUE TITLE
extends: only works with arrays

### ISSUE THREAD

**Issue by atli-c** [firecow/gitlab-ci-local#42]
`extends` only works with array, not with string.

```yaml
.base-demo:
    before_script: [echo "Demonstration"]
demo-extend-1:
    extends: [ .base-demo ]
    script: echo "This is '$CI_JOB_NAME' on GITLAB_CI='$GITLAB_CI'"
demo-extend-2:
    extends: [ demo-extend-1 ]
    variables:
        FOO: BAR
demo-extend-3:
    extends: demo-extend-1
    variables:
        FOO: BAZ
```

Expected output (and expected exit code: 0):
> Demonstration
> This is 'demo-extend-3' on GITLAB_CI='false'

Actual output (and actual exit code: 1):
> d is used by demo-extend-3, but is unspecified

gitlab-ci-local version: 4.6.0

**Comment by firecow** [comment]
Closing https://github.com/firecow/gitlab-ci-local/issues/40 in favor of this better described issue.

**Comment by firecow** [comment]
Fixed in 4.6.2
